src/smartcontracts/typesystem/managedDecimal.ts

```diff
@@ -38,7 +41,7 @@ export class ManagedDecimalValue extends TypedValue {
     }
 
     getPrecision(): number {
-        return this.value.toString(this.scale).replace(".", "").length;
+        return this.value.toFixed(this.scale).replace(".", "").length;
```

Comment:
Below, `==` is tricky for BigNumber values.

For example, the following code prints `false`:

```
const a = new BigNumber(100);
const b = new BigNumber(100);
console.log(a == b);
``` 

Use `eq()` instead?